packages/core-utils/src/provider.ts

```diff
@@ -24,17 +29,26 @@ export interface FallbackProviderConfig {
   weight?: number
 }
 
-export const FallbackProvider = (config: string | FallbackProviderConfig[]) => {
+export const FallbackProvider = (
+  config: string | FallbackProviderConfig[],
+  headers?: HttpHeaders
+) => {
   const configs = []
+  // Handle the case of a string of comma delimited urls
   if (typeof config === 'string') {
     const urls = config.split(',')
     for (const [i, url] of urls.entries()) {
+      const connectionInfo: ConnectionInfo = { url }
+      if (typeof headers === 'object') {
```

Comment:
what does this if statement protect against?

Reply:
It lets you not pass in the headers object without assigning `undefined` to the object, its more of a safety thing but i suppose the code is the same without it